### PR TITLE
readd support for Cluster.FeatureLayer and fix the plugin

### LINKED ISSUE #156: Uncaught TypeError: Cannot read property 'options' of undefined
Discovered during the release today, I get this error when trying to create a layer:

image
image

The last change to this function was [here](https://github.com/Esri/esri-leaflet-renderers/commit/b1679eb3ad2d0164fc1250c390642ed99b919475#diff-4c9ef9c3abfbeeb7f7027112c216ae50). @jgravois any ideas on this one?

### PR DESCRIPTION
**Pull request by jgravois**
relies on Esri/esri-leaflet-cluster#31
resolves #156 

while i was in there I also:
* started importing the ES modules from Leaflet and Esri Leaflet directly (good for webpack users etc.)
* moved all the pages we created back in the day for visual QA/QC out of `/spec` and into `/debug` so they'd be easier for folks to find.

the only thing i don't like about this approach is that AMD loaders like Dojo and RequireJS now require that the cluster plugin be loaded whether its needed or not.

neither are a popular option outside of JSAPI land these days though. i can live with the extra cruft since folks using modern bundlers or the UMD won't have to worry about it.

the diff is a bit of a mess, so don't be shy if you have any questions.

### CODE REVIEW

File: src/FeatureLayerHook.js [hunk 1]
```diff
@@ -8,9 +9,11 @@ function wireUpRenderers () {
...
     return;
   }
-  var oldOnAdd = L.Util.bind(this.onAdd, this);
-  var oldUnbindPopup = L.Util.bind(this.unbindPopup, this);
-  var oldOnRemove = L.Util.bind(this.onRemove, this);
+  var oldOnAdd = Util.bind(this.onAdd, this);
+  var oldUnbindPopup = Util.bind(this.unbindPopup, this);
+  var oldOnRemove = Util.bind(this.onRemove, this);
+
+  L.Util.bind(this.createNewLayer, this);
```
Comment (jgravois) [inline src/FeatureLayerHook.js:16]: right now this throws an error. fixed by https://github.com/Esri/esri-leaflet-cluster/pull/31.

i couldn't find a layer to test with, but it looks like `this.createNewLayer` is only really useful for displaying polygon services rendered using proportional marker symbols.

**Comment by gavinr** [comment]
Thanks @jgravois! I've merged https://github.com/Esri/esri-leaflet-cluster/pull/31 and released under v2.1.0, will test/merge/release this PR soon.

**Comment by jgravois** [comment]
very cool!

**Comment by gavinr** [comment]
These fixes were released in https://github.com/Esri/esri-leaflet-renderers/releases/tag/v2.1.1. Thanks again @jgravois

**Comment by francharbo** [comment]
Sorry to reopen this, maybe I should create a bug ?

I tried to use v2.1.1 using import style declaration for the plugins : 
```js
import L from 'leaflet'
import * as esri from 'esri-leaflet'
import * as Cluster from 'esri-leaflet-cluster'
import 'leaflet.markercluster/dist/leaflet.markercluster'
import MarkerClusterCss from 'leaflet.markercluster/dist/MarkerCluster.css'
import DefaultMarkerCss from 'leaflet.markercluster/dist/MarkerCluster.Default.css'

L.esri = esri
L.esri.Cluster = Cluster

import 'esri-leaflet-renderers'
```

But the initHook aren't attached to esri and esri.Cluster featureLayer.

It does work with v2.1.0 though.
Any idea of how to solve this ?

**Comment by jgravois** [comment]
> Any idea of how to solve this?

off the top of my head no, but if you're using import statements, why not go whole hog and just import the ES Modules you're using?

```js
import { map } from 'leaflet'
import { featureLayer } from 'esri-leaflet'
```

either way if you're able to provide a complete, self-contained, simplified, ready to run repro case instead of just a code snippet, it'd speed up the debugging process.

**Pull request merged**